file_organizer_v2/src/file_organizer/services/deduplication/embedder.py

```
@@ -0,0 +1,317 @@
+                min_df=min_df,
+                max_df=max_df,
+                stop_words='english',
+                lowercase=True,
+                strip_accents='unicode'
+            )
+
+        except ImportError:
+            raise ImportError(
+                "scikit-learn is required for document embedding. "
+                "Install with: pip install scikit-learn>=1.4.0"
+            )
```

Ruff (0.14.13)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)